architecture/core.rst

```diff
@@ -390,6 +390,145 @@ normal world supports scheduling of processes. Even on UP systems, supporting
 several trusted threads in optee_os helps normal world scheduler to be
 efficient.
 
+Core handlers for native interrupts
+===================================
+
+OP-TEE core provides methods for device drivers to setup and register
+handler functions for native interrupt controller drivers
+(see:ref:`native_foreign_irqs`).
+Interrupt handlers can be nested for when an interrupt controller
+exposes interrupts which signaling is multiplexed on an interrupt
+controlled by a parent interrupt controller.
+
+Interrupt controllers are represented by an instance of ``struct itr_chip``.
+An interrupt controller exposes a given number of interrupts, each identified
+by an index from 0 to N-1 where N is the total number of interrupts exposed
+by that controller. In the literature, an interrupt index identifier
+is called interrupt number.
+
+**Interrupt management API functions**
+
+Interrupt management resources are declared in header file
+kernel/interrupt.h_. Interrupt consumers main API functions are:
+
+    - ``interrupt_enable()`` and ``interrupt_disable()`` to respectively
+      enable and disable an interrupt.
+
+    - ``interrupt_mask()`` and ``interrupt_unmask()`` to respectively
+      mask/unmask an interrupt.
```

Comment:
mask or unmask

I think we need to mention the difference between mask/unmask and enable/disable.

Reply:
ok, i'll add a comment.